src/lib/y2network/fake_interface.rb

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+
+module Y2Network
+  # This class represents an interface which does not exists in the system
```

Comment:
Then why does it exist here? For example:
> It's because each {Foo} must have an {Interface} so when {Gandalf} does not provide one we use a {FakeInterface}

Reply:
Main reason is hot-plug devices. So it is physical devices that is not physically attached. I am not sure about virtual devices built from fake one, if it is also fake or not. ( I expect yes, as it is not on system till underlaying device is attached )